agent/config/builder.go

```diff
@@ -942,6 +943,21 @@ func (b *Builder) Validate(rt RuntimeConfig) error {
 	if err := structs.ValidateMetadata(rt.NodeMeta, false); err != nil {
 		return fmt.Errorf("node_meta invalid: %v", err)
 	}
+	// Handle deferred gossip encryption key delivery
+	if rt.EncryptKeyFromAPI {
+		if rt.EncryptKey != "" {
+			b.warn("encrypt key will be ignored when using encrypt_key_from_api")
```

Comment:
Maybe just hard fail here to drive home "please don't put secrets on disk if you don't want secrets on disk"

Reply:
Trying to think through if there is any reason you might want this as an interim option during transition but I suppose not. Yeah I think that's reasonable.